src/view/DisplayRequestsView.java

```diff
@@ -0,0 +1,88 @@
+
+import entity.User;
+
+public class DisplayRequestsView extends JPanel implements ActionListener, PropertyChangeListener {
+
+    public final String viewname = "Display Requests";
+    private final DisplayRequestsViewModel displayRequestsViewModel;
+    private final DisplayRequestsController displayRequestsController;
+    private final DisplayProfileController displayProfileController;
+    private final LoggedInViewModel loggedInViewModel;
+
+    public DisplayRequestsView(DisplayRequestsViewModel displayRequestsViewModel, DisplayRequestsController displayRequestsController,
+                               DisplayProfileController displayProfileController, LoggedInViewModel loggedInViewModel) {
+        this.displayRequestsViewModel = displayRequestsViewModel;
+        this.displayRequestsController = displayRequestsController;
+        this.displayProfileController = displayProfileController;
+        this.loggedInViewModel = loggedInViewModel;
+
+        JLabel title = new JLabel(DisplayRequestsViewModel.TITLE_LABEL);
+        title.setAlignmentX(Component.CENTER_ALIGNMENT);
+
+        JPanel buttons = new JPanel();
+
+
+        // Adding view profile and accept request buttons for every request
+        DisplayRequestsState currentState = displayRequestsViewModel.getState();
+        ArrayList<String> requests = currentState.getRequests();
+        for (String request : requests) {
+            JLabel requestUsername = new JLabel(request);
+            this.add(requestUsername);
+
+            JButton view = new JButton(DisplayRequestsViewModel.VIEW_BUTTON_LABEL);
+            buttons.add(view);
+
+            JButton accept = new JButton(DisplayRequestsViewModel.ACCEPT_BUTTON_LABEL);
+            buttons.add(accept);
+
+            view.addActionListener(
+                    new ActionListener() {
+                        public void actionPerformed(ActionEvent evt) {
+                            if (evt.getSource().equals(view)) {
+                                displayProfileController.execute();
+                            }
+                        }
+                    }
+            );
+
+            accept.addActionListener(
+                    new ActionListener() {
+                        public void actionPerformed(ActionEvent evt) {
+                            if (evt.getSource().equals(accept)) {
+
+                                acceptRequestController.execute(); // Ryan's use case
+                            }
+                        }
+                    }
+            );
+            this.setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
+
+            this.add(title);
+            this.add(buttons);
+        }
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent evt) {
+        JOptionPane.showConfirmDialog(this, "You have accepted this request.");
+    }
+
+    @Override
+    public void propertyChange(PropertyChangeEvent evt) {
+        //does not display anything for either state...
```

Comment:
Maybe we need to set the state (set username) for the AcceptRequests use case so that we can keep track of the current user.